web/src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -27,12 +28,14 @@ public GiftCertificateController(GiftCertificateService giftCertificateService,
     }
 
     @GetMapping(value = "/{id}")
+    @PreAuthorize("hasAnyAuthority('ROLE_USER','ROLE_ADMIN')")
```

Comment:
Seems to me you missed this requirement: 
- Guest:
    * Read operations for main entity.
    
It means that Guest users should be able to see the gift certificates

Reply:
I got you. I will fix it.